core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -122,11 +116,14 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     override def fromConfigObject(co: ConfigObject, default: Option[V] :: U): Try[FieldType[K, V] :: T] = {
       val keyStr = mapping(key.value.toString().tail)
       for {
-        v <- improveFailure(
-          vFieldConvert.value.from(co.get(keyStr)) match {
-            case Failure(CannotConvertNullException) =>
+        v <- improveFailure[V](
+          (co.get(keyStr), vFieldConvert.value) match {
+            case (null, converter: OptionConfigConvert[_]) =>
```

Comment:
Even though handling `null` isn't ideal, I don't like having the `from` method receiving an `Option[ConfigValue]`, as most of the time custom implementations will only be handling `Some` values, but would still have to do something about the `None`. 

One alternative would be for `ConfigConvert` to have a `onMissingKey: Option[T]` method that would be `None` by default. That would allow customizing the behavior for missing keys while not breaking the API and not having the `from` method called with `null`.